src/components/AvroToTypescriptCompiler.ts

```diff
@@ -1,40 +1,68 @@
+import * as fs from "fs";
+import * as path from "path";
 import {BaseCompiler} from "../core/BaseCompiler";
-import {FileHelper} from "../helpers/FileHelper";
+import {DirHelper} from "../helpers/DirHelper";
 import {TypeHelper} from "../helpers/TypeHelper";
-import {AvroSchemaInterface} from "../interfaces/AvroSchemaInterface";
-import {AvroSchemaConverter} from "./avroToTypescript/AvroSchemaConverter";
+import {AvroSchemaInterface, RecordType} from "../interfaces/AvroSchemaInterface";
+import { CompilerOutputInterface } from "../interfaces/CompilerOutputInterface";
+import {ExportModel} from "../models/ExportModel";
+import {RecordConverter} from "./avroToTypescript/RecordConverter";
 
 export class AvroToTypescriptCompiler extends BaseCompiler {
     public tsSchemaContent: string;
+    public exports: ExportModel[];
 
-    public async compile(): Promise<void> {
-
-        if (this.isCompileReady() === false) {
-            this.addError(AvroToTypescriptCompiler.errorMessage.notCompileReady);
-            throw new Error(AvroToTypescriptCompiler.errorMessage.notCompileReady);
+    public async compileFolder(): Promise<void> {
+        try {
+            fs.readdir(this.avroSchemaPath, async (err, files) => {
+                for (const file of files) {
+                    try {
```

Comment:
This function seems too nested. Why is there a try inside a try block and the catch blocks are the same? Is there a difference if you leave out the inner try block? I think the outer catch would catch it and log the error. 

I would extract the for block in a separate function `compileFile` to reduce complexity.